src/main/java/org/thoughtcrime/securesms/notifications/NotificationCenter.java

```diff
@@ -387,26 +393,33 @@ public void notifyWebxdc(int accountId, int contactId, int msgId, String text) {
 
         DcContext dcContext = context.dcAccounts.getAccount(accountId);
         DcMsg dcMsg = dcContext.getMsg(msgId);
-        JSONObject info;
+        DcMsg parentMsg;
         if(dcMsg.getType() == DcMsg.DC_MSG_WEBXDC) {
-          info = dcMsg.getWebxdcInfo();
-        } else { // info message, get from parent xdc
-          info = dcMsg.getParent() != null? dcMsg.getParent().getWebxdcInfo() : new JSONObject();
+            parentMsg = dcMsg;
+        } else { // info message, get parent xdc
+            parentMsg = dcMsg.getParent() != null? dcMsg.getParent() : dcMsg;
+        }
+
+        if (Util.equals(visibleWebxdc, new Pair<>(accountId, parentMsg.getId()))) {
+            return; // do not notify if the app is already open
         }
+
+        JSONObject info = parentMsg.getWebxdcInfo();
         final String name = JsonUtils.optString(info, "name");
         String shortLine = name.isEmpty()? text : (name + ": " + text);
-        maybeAddNotification(accountId, dcContext.getChat(dcMsg.getChatId()), msgId, shortLine, shortLine, false);
+        DcChat dcChat = dcContext.getChat(dcMsg.getChatId());
+        maybeAddNotification(accountId, dcChat, msgId, shortLine, shortLine, false, dcChat.isMultiUser());
```

Comment:
how is that "mentioning" supposed to work?

if i understand the code correctly, that would result in _any_ webxdc-notification to  be treated as "mention" and bypass group-muting

Reply:
that was the initial idea, but that was before "*" came in, oth, notifying all even if group is muted is still nice, and is similar to what happen in a muted group when someone uses the `@all` mention

Reply:
I think that for webxdcs, `isMention` should either always be `false` or `wasDirectlyNotified` (i.e. not only with `*`). Because I think it would be quite annoying to get notifications `XY added an event to the calendar` all the time even though you muted the group, and look like a bug.

Reply:
what we could do to mitigate potential annoyances is to have a setting in notifications to disable notification of "mentions" and webxdc in muted chats, like I did for ArcaneChat:

![image](https://github.com/user-attachments/assets/118c97f2-16b2-483c-b906-b8f45a8abe88)

Reply:
2024? he is moving forward since the 80s

Reply:
> Because I think it would be quite annoying to get notifications `XY added an event to the calendar all the time` even though you muted the group, and look like a bug.

agree, we should not do that.

in general, not sure to introduce these "mentions" last minute that the user cannot get rid of.

an option might be nice, but we should have that on all UI then, this is nothing for today.

Reply:
> in general, not sure to introduce these "mentions" last minute that the user cannot get rid of.

the webxdc notifications in muted groups was intended since the very beginning or that was hpk told me, I am fine with not doing it, then notifications for webxdc will be lost since there is no way to read them later unlike with messages